### DISCUSSION TITLE
Color coding Sectors

### DISCUSSION THREAD

**Discussion by Veracity2693** [jokergoo/circlize#344] [Q&A]
Hi,

Thank you for a wonderful package for visualization and making it public for everyone's use.

I have come across an error that I am looking for some help on.

I tried using ColorRamp2 to give a range of colors to sectors but it gives the following error:
Error in rep(grid.col, n) : 
  attempt to replicate an object of type 'closure'

Pardon me for my beginner-level knowledge in R.

**Comment by jokergoo** [comment]
Can you tell me how you use the function? Maybe you can paste your full set of code here.

**Comment by Veracity2693** [comment]
id | size
-- | --
Reg. of cell populationproliferation | 23.75742587
Pos. reg. of gene expression | 21.46836412
Pos. reg. of developmentalproc. | 21.46836412
Pos. reg. of RNA metabolicproc. | 22.66607561
Epithelium development | 21.46836412
Reg. of multicellularorganismal development | 21.46836412
Pos. reg. of cell populationproliferation | 20.13798421
Tissue morphogenesis | 18.63679501
Animal organ morphogenesis | 20.13798421
Reg. of protein modificationproc. | 21.46836412
Pos. reg. of transcription,DNA-templated | 21.46836412
Pos. reg. of nucleicacid-templated transcription | 21.46836412
Pos. reg. of RNA biosyntheticproc. | 21.46836412
Reg. of transferase activity | 19.15922786
Reg. of kinase activity | 18.63679501
Cytokine-mediated signalingpathway | 19.15922786
Reg. of protein kinaseactivity | 18.0891648
Pos. reg. of molecularfunction | 21.46836412
Epithelial celldifferentiation | 18.63679501

This is the df2 data frame and the corresponding values which I tried making a color ramp with and used the code as following


library(circlize)

# Or use a data frame
df = read.table("Chord_GO_KEGG.txt",sep='\t', header = TRUE)
df2 = read.table("SectorValues.txt",sep='\t', header = TRUE)

grid.col = colorRamp2(range(df2["size"]), c("#004bff", "#bfd2ff"), transparency = 0.1)

col_fun = colorRamp2(range(df["value"]), c("#FFEEEE", "#FF0000"), transparency = 0.1)

#chordDiagram(df, grid.col = grid.col, col = col_fun)

png("Chord_Diagram_GO_KEGG.png", units = "in", width = 15.5, height = 16.5, res=1200) 

chordDiagram(df, grid.col = grid.col, col = col_fun, annotationTrack = c("grid"), annotationTrackHeight = mm_h(10), link.border="white", link.zindex = rank(df["value"]), link.sort = TRUE, link.decreasing = TRUE)

dev.off()
circos.track(track.index = 1, panel.fun = function(x, y) {
  circos.text(CELL_META$xcenter, CELL_META$ylim[1], CELL_META$sector.index, 
              facing = "reverse.clockwise", niceFacing = TRUE, adj = c(0, 0.5))}, bg.border = NA)

circos.clear()

If I need to post other files, please let me know. And thank you again for trying to help me

**Comment by jokergoo** [comment]
```r
chordDiagram(df, grid.col = grid.col, ...
```

Argument `grid.col` can only be set to a vector of colors. The object `grid.col` you generated with `colorRamp2()` is a color mapping function. Maybe you can do:

```r
grid.col = colorRamp2(range(df2["size"]), c("#004bff", "#bfd2ff"), transparency = 0.1)(df2["size"])
names(grid.col) = df2[, 1]. ## now grid.col is a named vector
chordDiagram(df, grid.col = grid.col, ...
```
